fdbserver/RestoreApplier.actor.h

```diff
@@ -66,14 +230,61 @@ struct ApplierBatchData : public ReferenceCounted<ApplierBatchData> {
 	void addref() { return ReferenceCounted<ApplierBatchData>::addref(); }
 	void delref() { return ReferenceCounted<ApplierBatchData>::delref(); }
 
-	explicit ApplierBatchData(UID nodeID, int batchIndex) : counters(this, nodeID, batchIndex) {
+	explicit ApplierBatchData(UID nodeID, int batchIndex)
+	  : counters(this, nodeID, batchIndex), applyStagingKeysBatchLock(SERVER_KNOBS->FASTRESTORE_APPLYING_PARALLELISM) {
 		pollMetrics =
 		    traceCounters("FastRestoreApplierMetrics", nodeID, SERVER_KNOBS->FASTRESTORE_ROLE_LOGGING_DELAY,
 		                  &counters.cc, nodeID.toString() + "/RestoreApplierMetrics/" + std::to_string(batchIndex));
 		TraceEvent("FastRestoreApplierMetricsCreated").detail("Node", nodeID);
 	}
 	~ApplierBatchData() = default;
 
+	void addMutation(MutationRef m, Version ver) {
+		if (!isRangeMutation(m)) {
+			if (stagingKeys.find(m.param1) == stagingKeys.end()) {
+				stagingKeys.emplace(m.param1, StagingKey());
+			}
+			stagingKeys[m.param1].add(m, ver);
```

Comment:
you can use an iterator to avoid another lookup in 247. I.e.,

```
auto it = stagingKeys.find(m.param1);
if (it == stagingKeys.end()) {
  bool inserted;
  std::tie(it, inserted) = stagingKeys.emplace(m.param1, StagingKey());
}
it->second.add(m, ver);
```